app/views/shell/_top.html.erb

```diff
@@ -58,6 +58,7 @@
           <%= yield(:audio) %>
         </div>
         <%= render "layouts/top_bar" %>
+        <div id="active-broadcast" class="broadcast-wrapper"></div>
```

Comment:
Do we want this empty DIV to always be there even if there's no announcement visible? Is there any way to avoid that?

Reply:
Also, we do have something like `crayons-banner` (not documented so it's absolutely fair you weren't aware of that :D) that basically supposed to be this kind of a banner... We do have several variations:

```
<div class="crayons-banner">...</div>
```
![image](https://user-images.githubusercontent.com/108287/84005078-d7c81e80-a96c-11ea-8301-f649ceb0fca8.png)


```
<div class="crayons-banner crayons-banner--brand">...</div>
```
![image](https://user-images.githubusercontent.com/108287/84005174-fc23fb00-a96c-11ea-82cb-02bc3b174958.png)

```
<div class="crayons-banner crayons-banner--success">...</div>
```
![image](https://user-images.githubusercontent.com/108287/84005320-3d1c0f80-a96d-11ea-9f00-0c8a2c9897bd.png)

```
<div class="crayons-banner crayons-banner--warning">...</div>
```
![image](https://user-images.githubusercontent.com/108287/84005355-4dcc8580-a96d-11ea-8e54-beb69f5e8f45.png)

```
<div class="crayons-banner crayons-banner--error">...</div>
```
![image](https://user-images.githubusercontent.com/108287/84005406-62108280-a96d-11ea-87b3-a7df155773f7.png)



We could obviously add additional ones and/or expend that component if needed.

Reply:
@ludwiczakpawel:
> Do we want this empty DIV to always be there even if there's no announcement visible? Is there any way to avoid that?

The only point where we know whether an announcement exists or not is when we fetch it [and assign it from the backend](https://github.com/thepracticaldev/dev.to/pull/8301/files#diff-c9d5cb67f974ae68d44442f67dea97f2R41). When we initialize the page, we check to see if any data [was assigned on fetch](https://github.com/thepracticaldev/dev.to/pull/8301/files#diff-a99b7a23c274fa6b57e161e372ad2e28R24), and that's when we do something with the `<div id="active-broadcast" class="broadcast-wrapper"></div>`. The only other option here is to actually insert the div _into_ the body using `append()`, but I personally find that more confusing and a vector for potential UI bugs: we'd have to know _where_ in the body to append it, and we'd be relying on no other elements in the body being shifted around or accidentally added/removed, otherwise the `<div>` could render in a completely incorrect location. Also, it's worth mentioning that the `<div id="message-notice"></div>` element already exists (and is empty except for when used for chat channels), and it would be odd to do something entirely different to that, I think. 


> Also, we do have something like crayons-banner (not documented ...

Yeah, I was chatting with @lisasy about this! Since our `/internal` pages don't have access to crayons, the "preview" on `/internal/broadcasts/:id/edit` does not render any crayons variables. I can definitely add it, but I don't think that this blocks the PR, and would prefer to do it in a separate PR to avoid this one from getting big :)

Reply:
Thanks, that was very helpful explanation :) Even though I don't like empty containers 🙈